iroha_p2p/src/network.rs

```diff
@@ -86,8 +107,17 @@ where
         self.broker.subscribe::<Received<T>, _>(ctx);
         // from other iroha subsystems
         self.broker.subscribe::<Post<T>, _>(ctx);
-        // from listener
-        ctx.notify_with_context(self.listener_stream());
+        // Start listener
+        match self.listen().await {
+            Ok(_) => {
+                // register for peers from listener
+                ctx.notify_with_context(self.listener_stream());
+            }
+            Err(e) => {
+                // TODO send some message to Sumeragi that we couldn't start
+                warn!(%e, "Error starting listener!");
```

Comment:
That is an error though

Reply:
Moved TcpListener start to constructor. Error will bubble up.